impeller/compiler/code_gen_template.h

```diff
@@ -173,7 +173,14 @@ std::move({{ arg.argument_name }}){% if not loop.is_last %}, {% endif %}
   // ===========================================================================
   // Metadata for Vulkan =======================================================
   // ===========================================================================
-  static constexpr std::array<DescriptorSetLayout,{{length(buffers)+length(sampled_images)}}> kDescriptorSetLayouts{
+  static constexpr std::array<DescriptorSetLayout,{{length(buffers)+length(sampled_images)+length(subpass_inputs)}}> kDescriptorSetLayouts{
+{% for subpass_input in subpass_inputs %}
+    DescriptorSetLayout{
+      {{subpass_input.binding}}, // binding = {{subpass_input.binding}}
```

Comment:
I think its a bit of a mistake to assume that all bindings are in the set index zero. We should fix that separately as that affects all instances of `DescriptorSetLayout` in the generated header. For example, if we were to manually specify `layout(set=2, binding=n), we'd keel over.

In the same vein, I think its a mistake to assume that we can only have one input attachment and at index zero. I'd also put the input_attachment_index in the descriptor set layout. [The previous attempt at this patch did the same](https://github.com/flutter/engine/commit/6f223db0fc2729e24b9c255186c2766d89a8c056).

You should be able to reproduce this by manually specifying the input_attachment_index to anything other than zero and checking the validity of the setup.

For the descriptor set index, we can file a separate followup issue.